pkg/errors/errors_test.go

```diff
@@ -0,0 +1,133 @@
+package errors
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestError_Error(t *testing.T) {
```

Comment:
this doesn't test the case of `e.Err = e` I don't think. 

Either we need to add the test case or just remove the check in the `Error()` method. I'm slightly partial to just removing it as I don't think it really saves anything.

Reply:
I also don't think we need to test it . that condition is there to avoid infinite recursion as it happened in one of core/app tests, I added a comment for more clarity.
```go 
if e.err == e {
	return e.message
}
```

Reply:
Can you link to the infinite recursion issue if there is one? 

We should at least update to `errors.Is` for the following lint errror

```
comparing with == will fail on wrapped errors. Use errors.Is to check for a specific error (errorlint)
```

Reply:
let me have a closer look at that lint complain. I was also thinking to add an `Is` method to the errors struct which will be handy for users.

Reply:
I did not get the same output from my linter with default rules, can you please tell me what rules did you use? so I can reproduce it?

Reply:
run `make lint` from main with the latest version of golangci-lint. 
The error might not pop up for you, because it is coming from `errCheck` which I ignored for now because there were too many lint errors to address during the initial PR. 

If you enable `errCheck` this should pop up. 

either way you should update to us `errors.Is` because `==` is not a sufficient check for error comparison.